Rename WebPageProxy and LocalFrameLoaderClient functions to be less ambiguous

rreno opened the pull request:
#### 8972ce9795a840738e367ae77a0ba98a0f639295
<pre>
Rename WebPageProxy and LocalFrameLoaderClient functions to be less ambiguous
<a href="https://bugs.webkit.org/show_bug.cgi?id=278597">https://bugs.webkit.org/show_bug.cgi?id=278597</a>
<a href="https://rdar.apple.com/134603009">rdar://134603009</a>

Reviewed by Chris Dumez.

didSameDocumentNavigationForFrameViaJSHistoryAPI can be called whenever
we are doing a same-document navigation such as when an anchor
element&apos;s click event handler navigates us to another part of the
document. This patch is a rename to the functions, log messages, and
IPC messages to hopefully be more clear that this represents any JS initiated same
document navigation not only same document navigations via the history
API.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didSameDocumentNavigationForFrameViaHistoryAPIOrJS):
(WebKit::WebPageProxy::didSameDocumentNavigationForFrameViaJSHistoryAPI): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaHistoryAPIOrJS):
(WebKit::WebLocalFrameLoaderClient::dispatchDidPushStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReplaceStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidPopStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:

Canonical link: <a href="https://commits.webkit.org/283623@main">https://commits.webkit.org/283623@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30416feb080993176191b7b2f5aaac733ea08020

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/66661 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46036 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/70695 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/17794 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/53835 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/17559 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53412 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12003 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/69728 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42384 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/57685 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34081 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39055 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16148 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60971 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15417 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72397 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/10618 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14779 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60742 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/10650 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/57746 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61076 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14764 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8737 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2356 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/41843 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42920 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44103 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/42663 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (changes requested):
I think this old name was more accurate?

rreno committed:
Rename WebPageProxy and LocalFrameLoaderClient functions to be less ambiguous
https://bugs.webkit.org/show_bug.cgi?id=278597
rdar://134603009

Reviewed by Chris Dumez.

didSameDocumentNavigationForFrameViaJSHistoryAPI can be called whenever
we are doing a same-document navigation such as when an anchor
element's click event handler navigates us to another part of the
document. This patch is a rename to the functions, log messages, and
IPC messages to hopefully be more clear that this represents any JS initiated same
document navigation not only same document navigations via the history
API.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didSameDocumentNavigationForFrameViaHistoryAPIOrJS):
(WebKit::WebPageProxy::didSameDocumentNavigationForFrameViaJSHistoryAPI): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaHistoryAPIOrJS):
(WebKit::WebLocalFrameLoaderClient::dispatchDidPushStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidReplaceStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::dispatchDidPopStateWithinPage):
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.h:

Canonical link: https://commits.webkit.org/283623@main
